server/src/utils/middlewares/locks.ts

```diff
@@ -0,0 +1,38 @@
+import { ExpectedError } from '@labrute/core';
+import { Request, Response, NextFunction } from 'express';
+import sendError from '../sendError.js';
+
+interface Locks {
+  [key: string]: boolean
+}
+
+const locks: Locks = {};
+
+export default function lockMiddleware(req: Request, res: Response, next: NextFunction) {
+  const { method, path } = req;
+
+  const { headers: { authorization } } = req;
+
+  if (authorization) {
+    const [id, token] = Buffer.from(authorization.split(' ')[1], 'base64')
+      .toString()
+      .split(':');
+
+    if (!id || !token || id === 'null' || token === 'null') {
+      return sendError(res, new ExpectedError('Invalid authorization header content'));
+    }
+
+    const key: string = `${method}:${path}:${id}:${token}`;
```

Comment:
You should avoid force typing as much as possible.  
There is also no need for the token in the key.

```suggestion
    const key= `${method}:${path}:${id}`;
```